### DISCUSSION TITLE
React component vs script tag in Nextjs application

### DISCUSSION THREAD

**Discussion by jkohlin** [orestbida/cookieconsent#292] [Q&A] [answered]
In a Nextjs application, what would be the implication of just adding cookieconsent in a script tag in Header.js?
Why should I implement it as a component?
```
export const Header = () => {
    return (
        <Head>
            <title>{title}</title>
            <meta name="description" content={description} />
            <link rel="canonical" href={canonical} />
            <script src="../utils/cookieconsent.js"></script>
             <script src="../utils/cookieconsent-config.js"></script>
        </Head>
    )
}
```

**Comment by orestbida** [comment, marked as answer]
In this case you'd make 2 or 3 additional requests (if we also count the stylesheet) which could have been avoided.
You'd also have to add the `defer` attribute to make sure that the `<body>` element actually exists.  

Using a react component allows you to bundle the scripts + stylesheet and save unnecessary HTTP requests.
